CodeGeneration/Sources/SyntaxSupport/GrammarGenerator.swift

```diff
@@ -60,4 +69,20 @@ struct GrammarGenerator {
       .map { " - `\($0.varOrCaseName)`: \(generator.grammar(for: $0))" }
       .joined(separator: "\n")
   }
+
+  /// Generates the list of possible token kinds for this particular ``Child``.
+  /// The child must be of kind ``ChildKind/token``. Otherwise, an empty string will be returned.
+  static func childTokenChoices(for child: Child) -> String {
+    let generator = GrammarGenerator()
+
+    if case .token(let choices, _, _) = child.kind {
+      if choices.count == 1 {
+        return " \(generator.grammar(for: choices.first!, backticked: false))"
+      } else {
+        return choices.map { " - \(generator.grammar(for: $0, backticked: false))" }.joined(separator: "\n")
+      }
```

Comment:
Instead of switching on `choices.count == 1` here again, what do you think about making the ternary operator in `Child.documentation` a full-grown `if` and then directly calling into `generator.grammar` from there? That would simplify `childTokenChoices` to always return a list.

Reply:
Makes sense to only have the if/else in any form only in one place instead of two — I'll take a look.